client/components/AllProducts-orig.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { fetchProducts } from '../store/products';
+import SearchBar from './SearchBar';
+import MapProducts from './MapProducts';
+import Row from 'react-bootstrap/Row';
+import CreateProduct from './CreateProduct';
+
+export class AllProducts extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: true,
+    };
+  }
+
+  componentDidMount() {
+    this.props.getProducts();
+    this.setState({ loading: false });
```

Comment:
Something like a loading state _has to be kept in redux_. getProducts is an asynchronous action, when our synchronous code in the componentDidMount gets to this line 20 here, the products have definitely not actually been loaded, and we're never going to have a loading status that actually reflects whether our data is loading or not